app/services/base_service.rb

```diff
@@ -201,7 +201,10 @@ def raise_if_error!
 
   def self.call(*, **, &)
     LagoTracer.in_span("#{name}#call") do
-      new(*, **).call(&)
+      instance = new(*, **)
+      return instance.call_with_audit_logs(&) if instance.audit_logs
```

Comment:
Maybe we could think at a generic name in case we need similar logic in the future? Something like `call_in_block` :thinking: